core/src/main/c/osx/kqueue.c

```diff
@@ -89,5 +91,39 @@ JNIEXPORT jint JNICALL Java_io_questdb_network_KqueueAccessor_kevent
                          &dontBlock);
 }
 
+JNIEXPORT jlong JNICALL Java_io_questdb_network_KqueueAccessor_pipe
+        (JNIEnv *e, jclass cl) {
+    int fds[2];
+    int res = pipe(fds);
+    if (res < 0) {
+        return res;
+    }
+    res = fcntl(fds[0], F_SETFL, O_NONBLOCK);
+    if (res < 0) {
+        return res;
+    }
+    return (jlong) fds[0] << 32 | (jlong) fds[1];
+}
 
+JNIEXPORT jint JNICALL Java_io_questdb_network_KqueueAccessor_readPipe
+        (JNIEnv *e, jclass cl, jint fd) {
+    char buf[1];
+    ssize_t s;
+    s = read((int) fd, buf, 1);
+    if (s != 1) {
+        return -1;
+    }
+    return (jint) buf[0];
+}
 
+JNIEXPORT jint JNICALL Java_io_questdb_network_KqueueAccessor_writePipe
+        (JNIEnv *e, jclass cl, jint fd) {
+    char buf[1];
+    ssize_t s;
+    buf[0] = 1;
+    s = write((int) fd, buf, 1);
```

Comment:
same

Reply:
Done in 1f11af7c55d62ed350bfdba5dffd5370f4fc17c8